File: angular-app/src/app/services/driver.service.ts
import { HttpHeaders } from '@angular/common/http';
import { Injectable } from '@angular/core';
import { Observable } from 'rxjs';
import { DriverDTO } from '../dto/driver-dto';
import { ApiService } from './api.service';

@Injectable({
  providedIn: 'root'
})
export class DriverService {
  
  private basUrl = "driver";

  private httpOptions = {
    headers: new HttpHeaders({
      "Content-Type": "application/json",
    })
  };
  
  constructor(private apiService: ApiService) {}
  
  getDriverList(): Observable<DriverDTO[]> {
    return this.apiService.get(`${this.basUrl}/all`);
  }

  createDriver(driverDTO: DriverDTO): Observable<Object> {
    return this.apiService.post(`${this.basUrl}/create`, driverDTO);
  }

  getDriverById(idDriver: number): Observable<DriverDTO>{
    return this.apiService.get(`${this.basUrl}/read/${idDriver}`);
  }

  updateDriver(idDriver:number, driverDTO: DriverDTO): Observable<Object>{
    return this.apiService.put(`${this.basUrl}/update/${idDriver}`, driverDTO);
  }

  deleteDriver(idDriver: number): Observable<Object>{
    return this.apiService.delete(`${this.basUrl}/delete/${idDriver}`);
  }
  
}

File: TransmiApp/TransmiApp/src/main/java/com/example/TransmiApp/service/DriverService.java
package com.example.TransmiApp.service;

import java.util.List;

import com.example.TransmiApp.dto.DriverDTO;
import com.example.TransmiApp.model.Driver;

public interface DriverService {
    List<Driver> getAllDrivers();
    List<Long> getDriverIds();
    DriverDTO getDriverById(Long idDriver);
    DriverDTO createDriver(DriverDTO driverDTO);
    DriverDTO updateDriver(Long idDriver, DriverDTO driverDTO);
    void deleteDriver(Long idDriver);
}
